driver-core/src/main/com/mongodb/internal/connection/SocketStreamHelper.java

```diff
@@ -63,6 +77,21 @@ static void initialize(final Socket socket, final InetSocketAddress inetSocketAd
         socket.connect(inetSocketAddress, settings.getConnectTimeout(MILLISECONDS));
     }
 
+    @SuppressWarnings("unchecked")
+    private static void setExtendedSocketOptions(final Socket socket) {
+        if (Arrays.stream(ExtendedSocketOptions.class.getDeclaredFields()).anyMatch(f -> f.getName().equals(TCP_KEEPCOUNT))) {
+            try {
+                socket.setOption((SocketOption<Integer>) ExtendedSocketOptions.class.getField(TCP_KEEPCOUNT).get(null),
```

Comment:
It's safer to call `setOption` reflectively as well, since that method was added in Java 9.  Otherwise, it's possible that this entire class will fail to load.  So something like:

                Method setOptionMethod = Socket.class.getMethod("setOption", SocketOption.class, Object.class);
                setOptionMethod.invoke(socket, ExtendedSocketOptions.class.getDeclaredField(TCP_KEEPCOUNT).get(null),
                                       TCP_KEEPCOUNT_LIMIT);

Reply:
Done